app/controllers/application_controller.rb

```diff
@@ -50,6 +51,13 @@ def require_subscription
     redirect_to root_path, notice: 'サブスクリプション登録が必要です。' unless current_user&.subscription?
   end
 
+  def require_scheduler_inheritation
+    return unless request.path_info.start_with?('/scheduler')
+    return if is_a?(SchedulerController)
+
+    head :internal_server_error
```

Comment:
```suggestion
    if request.path_info.start_with?('/scheduler') && !is_a?(SchedulerController)
      head :internal_server_error
    end
```

とした方がコードの意図が明確になりそう。

そして、できたらこの仕組みもテストコードがほしいけど、ちょっと難しいかなあ。。

Reply:
ガード節が好きでなんとなく使ってしまったのですが、たしかにこちらのほうが今回の意図を肯定形で表現できて分かりやすいですね。
ご提案いただいたコードだとrubocopにガード節か後置ifのどちらかにしろと言われたので、後置ifを用いた書き方に修正しました。
[782f084](https://github.com/fjordllc/bootcamp/pull/6970/commits/782f08408f2d867909f0728367c15296c85d1dab)

Reply:
この仕組みのテストコードは実現方法が思いつかずでして、そのままにしています:pray:

Reply:
了解です。最低限、一時的にcontrollerの親クラスを書き替えるなどして、手元で期待した動作をしているかどうかだけ確認しておいてほしいです〜。（スクショとかのエビデンスがあればなお良し）

Reply:
期待した動作が得られなかったので下記の修正を加えています。先に`require_active_user_login`が評価されて200が帰ってきていたため、`require_scheduler_inheritation`が先に評価されるように変更しました。
[b049748](https://github.com/fjordllc/bootcamp/pull/6970/commits/b049748e9f3f5c3ed3b35a15d82cd0e9c883fc33)

上記の変更を加えたうえで、下記の箇所で`NotifyCertainPeriodPassedAfterLastAnswerController`の親クラスを`ApplicationController`に一時的に書き換え、ディスクリプションに記載した方法で動作確認を行いました。
https://github.com/fjordllc/bootcamp/blob/b049748e9f3f5c3ed3b35a15d82cd0e9c883fc33/app/controllers/scheduler/daily/notify_certain_period_passed_after_last_answer_controller.rb#L3
その結果、500エラーとなることを確認しました。
![image](https://github.com/fjordllc/bootcamp/assets/46841037/3483aeb2-50e4-4bf2-8242-15850ca61849)
親クラスを`SchedulerController`に戻すと正常に動作することも確認しました。